### PR TITLE
Handle V1 indexes, some of which have unsorted posting offset tables.

### LINKED ISSUE #6535: Web UI - Error loading available metrics!
## Bug Report
**What did you do?**
Successfully upgraded Prometheus 2.0 to 2.15.0-rc0 and then open WEB UI.

**What did you expect to see?**
I expect /graph page without error and with working search suggest drop-down list.

**What did you see instead? Under which circumstances?**

- Open Web UI behind Apache proxy on https://prometheus.company.org/ via Google Chrome.
- I am forwarded to https://prometheus.company.org/graph page where I see "_Error loading available metrics!_" error.
- Search suggest drop-down list is **not** working, there are no suggestions.

**Environment**

* System information:

`Linux 3.10.0-1062.4.1.el7.x86_64 x86_64`

* Prometheus version:

```
prometheus, version 2.15.0-rc.0 (branch: HEAD, revision: 61f1953c50148b3857bad6ab9a5622e1f1277695)
  build user:       root@67bb57f8015b
  build date:       20191218-17:10:23
  go version:       go1.13.5
```

* Prometheus parameters:
```
/opt/prometheus/prometheus --storage.tsdb.retention.time=60d --storage.tsdb.retention.size=98GB --config.file=/opt/prometheus/prometheus.yml --storage.tsdb.path=/opt/prometheus/data --web.external-url=https://prometheus.company.org/ --web.console.templates=/opt/prometheus/consoles --query.timeout=1m --web.read-timeout=1m --web.route-prefix=/
```

### PR DESCRIPTION
**Pull request by brian-brazil**
Fixes #6535

Signed-off-by: Brian Brazil <brian.brazil@robustperception.io>

@bwplotka

### CODE REVIEW

File: tsdb/index/index.go [hunk 1]
```diff
@@ -1423,7 +1457,7 @@ func (r *Reader) LabelValues(names ...string) (StringTuples, error) {
 
 	skip := 0
 	for d.Err() == nil {
-		if skip == 0 {
+		if skip == 0 || true {
```
Comment (roidelapluie) [inline tsdb/index/index.go]: @brian-brazil mistake?

Comment (brian-brazil) [reply tsdb/index/index.go]: Test code that slipped through, I wanted to be sure I had one that was unsorted.

**Comment by roidelapluie** [comment]
@bwplotka TestReadIndexFormatV1 has been updated with those tests

**Comment by roidelapluie** [comment]
unless you make a difference between e2e and unit tests

**Comment by bwplotka** [comment]
Ah yes, sorry all good then. Cherry-picking commit with windows fix and this for 2.15.2 :+1:

**Pull request merged**